api/src/main/java/org/openmrs/module/mksreports/reports/OutpatientHistoryReportManager.java

```diff
@@ -0,0 +1,260 @@
+import org.openmrs.module.mksreports.common.Helper;
+import org.openmrs.module.mksreports.data.converter.ConceptDataTypeConverter;
+import org.openmrs.module.mksreports.data.converter.ConceptNameConverter;
+import org.openmrs.module.mksreports.data.converter.EncounterProviderFromIdConverter;
+import org.openmrs.module.mksreports.data.converter.EncounterTypeUUIDFromEncounterIdConverter;
+import org.openmrs.module.mksreports.data.converter.ObsProviderFromIdConverter;
+import org.openmrs.module.mksreports.data.converter.ObsValueFromIdConverter;
+import org.openmrs.module.mksreports.data.converter.VisitLocationFromIdConverter;
+import org.openmrs.module.mksreports.data.converter.VisitTypeFromIdConverter;
+import org.openmrs.module.mksreports.data.converter.VisitUUIDFromIdConverter;
+import org.openmrs.module.mksreports.data.obs.definition.ObsDatetimeDataDefinition;
+import org.openmrs.module.mksreports.dataset.definition.PatientHistoryEncounterAndVisitDataSetDefinition;
+import org.openmrs.module.mksreports.dataset.definition.PatientHistoryObsAndEncounterDataSetDefinition;
+import org.openmrs.module.mksreports.library.BasePatientDataLibrary;
+import org.openmrs.module.mksreports.library.EncounterDataLibrary;
+import org.openmrs.module.mksreports.library.ObsDataLibrary;
+import org.openmrs.module.reporting.common.ObjectUtil;
+import org.openmrs.module.reporting.common.SortCriteria;
+import org.openmrs.module.reporting.data.converter.DateConverter;
+import org.openmrs.module.reporting.data.converter.ObjectFormatter;
+import org.openmrs.module.reporting.data.encounter.definition.EncounterDataDefinition;
+import org.openmrs.module.reporting.data.encounter.definition.EncounterDatetimeDataDefinition;
+import org.openmrs.module.reporting.data.encounter.definition.EncounterIdDataDefinition;
+import org.openmrs.module.reporting.data.encounter.definition.EncounterTypeDataDefinition;
+import org.openmrs.module.reporting.data.obs.definition.ObsDataDefinition;
+import org.openmrs.module.reporting.data.obs.definition.ObsIdDataDefinition;
+import org.openmrs.module.reporting.data.patient.definition.PatientDataDefinition;
+import org.openmrs.module.reporting.data.patient.library.BuiltInPatientDataLibrary;
+import org.openmrs.module.reporting.dataset.definition.EncounterDataSetDefinition;
+import org.openmrs.module.reporting.dataset.definition.ObsDataSetDefinition;
+import org.openmrs.module.reporting.dataset.definition.PatientDataSetDefinition;
+import org.openmrs.module.reporting.evaluation.parameter.Mapped;
+import org.openmrs.module.reporting.evaluation.parameter.Parameter;
+import org.openmrs.module.reporting.report.ReportDesign;
+import org.openmrs.module.reporting.report.definition.ReportDefinition;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+@Component(MKSReportsConstants.COMPONENT_REPORTMANAGER_OPDHISTORY)
+public class OutpatientHistoryReportManager extends MKSReportManager {
```

Comment:
Let's rename this `PatientHistoryReportManager`.

Reply:
@mks-d , file renamed